app/src/main/java/org/cru/godtools/ui/tooldetails/ToolDetailsFragment.kt

```diff
@@ -76,6 +78,22 @@ class ToolDetailsFragment() :
         menu.setupPinShortcutAction()
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        when (requestCode) {
+            REQUEST_TUTORIAL_TIPS -> when (resultCode) {
+                Activity.RESULT_OK -> {
+                    settings.setFeatureDiscovered("$FEATURE_TUTORIAL_TIPS${dataModel.tipsTool}")
+                    dataModel.tipsTool?.let {
+                        tool -> dataModel.tipsType?.let {
+                        type -> dataModel.tipsLanguage?.let {
+                        language -> requireActivity().openToolActivity(tool, type, language, showTips = true) } } }
```

Comment:
I'm not a fan of nesting `let` blocks like this. a better pattern is:
```kotlin
val code = dataModel.tipsTool ?: return
val type = dataModel.tipsType ?: return
val language = dataModel.tipsLanguage ?: return
```

Reply:
instead of calling `openToolActivity` here, call `launchTrainingTips` with `force = true`

Reply:
Oh yeah, Elvis operator! I like that better and will change.